core/io/resource_loader.cpp

```diff
@@ -1188,6 +1194,36 @@ String ResourceLoader::get_resource_script_class(const String &p_path) {
 	return "";
 }
 
+String ResourceLoader::get_resource_editor_description(const String &p_path) {
+	String local_path = _validate_local_path(p_path);
+
+	for (int i = 0; i < loader_count; i++) {
+		if (loader[i]->has_editor_description_support()) {
+			String result = loader[i]->get_resource_editor_description(local_path);
+			if (!result.is_empty()) {
+				return result;
+			}
+		}
+	}
+
+	return "";
+}
+
+bool ResourceLoader::has_editor_description_support(const String &p_path) {
+	String local_path = _validate_local_path(p_path);
+
+	for (int i = 0; i < loader_count; i++) {
+		if (!loader[i]->recognize_path(local_path)) {
+			continue;
+		}
+		if (loader[i]->has_editor_description_support()) {
```

Comment:
Wouldn't it be better to

```suggestion
		return loader[i]->has_editor_description_support();
```
?

The current code is only useful in case where 2 loaders handle a resource, but only one has description support. That's rather unlikely I think?

Reply:
You do have a point there.

I wrote it like that so its behavior is consistent with `ResourceLoader::has_custom_uid_support()`:
https://github.com/godotengine/godot/blob/6e33e3b7af18b0e9913e22b33e3bb2dac8954ad0/core/io/resource_loader.cpp#L1204-L1217

Note that `ResourceLoader::get_resource_editor_description()` also looks for the first loader that returns true with `ResourceLoader::has_editor_description_support()`.

Let me know if performance is preferred here instead of consistency.